src/index.js

```diff
@@ -1,45 +1,195 @@
-   
-      `;
-};
-
-const displayItems = () => {
-  const todoList = [
-    {
-      description: 'Learn React',
-      completed: false,
-      index: 1,
-    },
-    {
-      description: 'complete the todo list',
-      completed: false,
-      index: 2,
-    },
-    {
-      description: 'Email Student success',
-      completed: false,
-      index: 3,
-    },
-
-  ];
-  const listItems = todoList;
-  listItems.forEach((item) => { addlistItem(item); });
-};
-// displayItems
-document.addEventListener('DOMContentLoaded', displayItems);
\ No newline at end of file
+class List {
+  constructor(description, index, completed) {
+    this.description = description;
+    this.index = index;
+    this.completed = completed;
+  }
+}
+
+// Create local storage
+
+if (localStorage.getItem('mylist') === null) {
+  localStorage.setItem('mylist', JSON.stringify([]));
+}
+const taskLs = JSON.parse(localStorage.getItem('mylist'));
+
+// updateLocalStorage
+
+function updateLocalStorage() {
+  localStorage.setItem('mylist', JSON.stringify(taskLs));
+  const another = JSON.parse(localStorage.getItem('mylist'));
+  return another;
+}
+
+// display todoList
+const displayItem = (items) => {
+  let todoos = '';
+
+  for (let i = 0; i < items.length; i += 1) {
+    const ischecked = items.completed == 'completed' ? 'checked' : '';
+
+    todoos += `
+  <div class="todoItem">
+  <div class="todoCheckbox">
+    <input type="checkbox" class="checkBox" ${ischecked} />
+    <p class="todoActivity" > ${items[i].description}  </p>
+  </div>
+  <div class="${items[i].index}">
+  <button class="addbtn">
+
+  <span class="edit" id='${items[i].index}'><i class="fa fa-pencil-square-o" aria-hidden="true"></i>
+  </span>
+  <span class="delete" id='${items[i].index}'>&cross;</span>
+  </button>
+  </div>
+  <button class="deleteItem">  </button>  
+  </div>   `;
+  }
+  return todoos;
+};
+
+// clear fields
+
+const clearFields = () => {
+  document.querySelector('.taskInput').value = '';
+};
+
+// showlist
+const showList = () => {
+  const listContent = document.querySelector('.list-content');
+  listContent.innerHTML = `
+
+ ${displayItem(taskLs)}`;
+
+  clearFields();
+};
+
+// Create new task
+const addItem = (description) => {
+  const anothertask = new List(description, taskLs.length + 1, false);
+  taskLs.push(anothertask);
+  updateLocalStorage();
+  showList();
+};
+// to edit
+
+const form = document.querySelector('.form');
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+  const task = document.querySelector('.taskInput').value;
+  addItem(task);
+});
+
+const deleted = (pop) => {
+  const removeLss = JSON.parse(localStorage.getItem('mylist'));
+  removeLss.forEach((element, index) => {
+    if (element.index == pop) {
+      removeLss.splice(index, 1);
+      localStorage.setItem('mylist', JSON.stringify(removeLss));
+
+      console.log(' ');
+    }
+  });
+  localStorage.setItem('mylist', JSON.stringify(removeLss));
+  location.reload();
+};
```

Comment:
- [ ] One of the project requirements is for you to reorder the `index` key value after deleting an item. Currently, when I delete an item, the `index` values are not reordered. All you have to do is use a `forEach` with `index` as its second parameter and set each `index` key value to `index + 1`. 👍